### ISSUE TITLE
Would it be possible to incorporate a boundary inside the active computational domain

### ISSUE THREAD

**Issue by arghyadeep-paul** [SmileiPIC/Smilei#556]
My problem requires that I have a spherical/circular body inside my (Cartesian) computational domain that behaves as an internal boundary. The boundary condition at this boundary is required to be `"remove"` for particles and `"open"` for EM fields (I understand that more sophisticated BCs at such an internal boundary present additional complications.)   

I have looked into the `Screen Diagnostics` , would it be maybe possible to modify the screen such that the particles crossing it are deleted and all fields inside it are set to zero at each iteration?? What i am wondering about is essentially some combination of the `PartWall` and the `DiagScreen`.

**Comment by mccoys** [comment]
We do have `Walls` already. Turning them into other shapes is not an impossible task. But I'm not sure when we will have time for that. This may also cause strange physics behavior.

**Comment by arghyadeep-paul** [comment]
Ok, I am not sure how much of a workload that would be for me. If it does not require in depth knowledge of the intricacies in the code, maybe I can try to do it myself. I am relatively new to Smilei though.

**Comment by mccoys** [comment]
You probably want to copy what is done with walls in the current version.

**Comment by arghyadeep-paul** [comment]
Could you please point me to the name of the source file?

**Comment by mccoys** [comment]
You can start [here](https://github.com/SmileiPIC/Smilei/blob/master/src/ParticleBC/PartWall.cpp) and

**Comment by arghyadeep-paul** [comment]
Yes. Thanks for the very prompt reply. I will study the code a bit and reply in this thread about what I plan to modify. It might take me some time as i am quite new to this code

**Comment by xxirii** [comment]
Hi,

To delete some particles on the fly is one of the tricky action in Smilei especially when the sorting is activated (in scalar or vectorized mode). We have already implemented some strategies to delete particles from physical mechanisms compatible with the sorting. Adapting them for the screening is probably the fastest solution. This can rapidly become a performance hotspot or a source of errors if not done properly. 

Basically, we usually first identify a list of Particles to delete (using a mask) and we then delete all leaving particles in an optimized manner.
